### PR TITLE
Add monster armor and strength options

### PR DESCRIPTION
**Pull request by RedShakespeare**
#### Summary

SUMMARY: Features"Add monster armor and strength options in worldoptions"
#### Purpose of change

Allow the players to modify the monsters' armor and strength(melee dice, grab strength, bash skill, etc.) in the world options in order to adjust the difficulty.
#### Describe the solution

Add two options "MONSTER_ARMOR" and "MONSTER_STRENGTH" and adjust monster data based on these two options.
#### Describe alternatives you've considered

List the data in detail and allow players to adjust everyone of them.

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -1949,6 +1949,16 @@ void options_manager::add_options_world_default()
          1, 1000, 100, COPT_NO_HIDE, "%i%%"
        );
 
+    add( "MONSTER_ARMOR", "world_default", translate_marker( "Monster armor" ),
+         translate_marker( "Determines how much damage monsters can defend.  A higher value makes monsters harder to get hurt and a lower makes them more vulnerable.  Requires world reset." ),
```
Comment (Night-Pryanik) [inline src/options.cpp]:
```suggestion
         translate_marker( "Determines how much damage monsters can block.  A higher value makes monsters harder to get hurt and a lower makes them more vulnerable.  Requires world reset." ),
```

**Comment by kevingranade** [comment]
This fits a lot better in the game balance json than in an option.

**Comment by RedShakespeare** [comment]
> This fits a lot better in the game balance json than in an option.

I thought of the game balance json as an alternative as well. However, as far as I have seen in cdda social network groups, most of the players prefer to change the difficulty in options conveniently to fits their own requirements instead of using a balance mod written by others, let alone writing json themselves since not every player understand json or even English. 

Besides, the existing game balance jsons usually only aim at a yes-no question such as whether the npcs have to consume foods, whether the dex attributes influence the speed, whether the clothes are filthy, etc., while existing linear adjustments about  game balance such as monster evaluation speed, monster speed and monster resilience nearly all lie in options. As for the thinking above, I assume that the changes in options may be a better choice.

File: src/options.cpp (hunk 1 above)
Comment (RedShakespeare) [reply src/options.cpp]: Thank you so much!

**Comment by kevingranade** [comment]
On second look, I'm not clear on how this is a meaningful tunable to add at all.

**Pull request closed without merging**